### PR TITLE
BugSplat integration

### PR DESCRIPTION
**Pull request by huffman**

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 27384dbe0a9fccb03ae65dbf1cefd1f937f0e8ba available at: https://gist.github.com/a8e1ed8dcc8614275f3d<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7052.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7052.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 20cf7c0099e55eff21e8070e835e25e3bb52346d available at: https://gist.github.com/a8e1ed8dcc8614275f3d<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7052.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7052.dmg

**Comment by huffman** [comment]
test this please

**Comment by huffman** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: 20cf7c0099e55eff21e8070e835e25e3bb52346d available at: https://gist.github.com/a8e1ed8dcc8614275f3d<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7052.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7052.dmg

**Comment by huffman** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: 20cf7c0099e55eff21e8070e835e25e3bb52346d available at: https://gist.github.com/a8e1ed8dcc8614275f3d<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7052.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7052.dmg

File: cmake/macros/AddBugSplat.cmake [hunk 1]
```diff
@@ -0,0 +1,20 @@
+macro(add_bugsplat)
+  find_package(BugSplat)
```
Comment (birarda) [inline cmake/macros/AddBugSplat.cmake]: wrap the find in a `if (NOT BUGSPLAT_CHECKED)` block and then set the global property once checked (same as TargetNsight.cmake does)

File: cmake/macros/AddBugSplat.cmake [hunk 2]
```diff
@@ -0,0 +1,20 @@
+macro(add_bugsplat)
```
Comment (birarda) [inline cmake/macros/AddBugSplat.cmake]: add header to this file

File: cmake/modules/FindBugSplat.cmake [hunk 3]
```diff
@@ -0,0 +1,19 @@
+if (WIN32)
```
Comment (birarda) [inline cmake/modules/FindBugSplat.cmake]: add header to this file

**Comment by huffman** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: 20cf7c0099e55eff21e8070e835e25e3bb52346d available at: https://gist.github.com/a8e1ed8dcc8614275f3d<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7052.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7052.dmg

File: cmake/macros/AddBugSplat.cmake (hunk 1 above)
Comment (huffman) [inline cmake/macros/AddBugSplat.cmake]: that was dumb of me considering I do read the property.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 3355019cbed8d6bd00b013bae931102ab3e28aca available at: https://gist.github.com/a8e1ed8dcc8614275f3d<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7052.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7052.dmg

File: cmake/macros/AddBugSplat.cmake [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+#  Distributed under the Apache License, Version 2.0.
+#  See the accompanying file LICENSE or http:#www.apache.org/licenses/LICENSE-2.0.html
+#
+
+macro(add_bugsplat)
+  get_property(BUGSPLAT_CHECKED GLOBAL PROPERTY CHECKED_FOR_BUGSPLAT_ONCE)
+
+  if (NOT BUGSPLAT_CHECKED)
+    find_package(BugSplat)
+    set_property(GLOBAL PROPERTY CHECKED_FOR_NSIGHT_ONCE TRUE)
```
Comment (birarda) [inline cmake/macros/AddBugSplat.cmake]: this should be CHECKED_FOR_BUGSPLAT_ONCE

File: interface/src/main.cpp [hunk 5]
```diff
@@ -25,8 +25,20 @@
...
+#endif
+
 int main(int argc, const char* argv[]) {
     disableQtBearerPoll(); // Fixes wifi ping spikes
+
+#if HAS_BUGSPLAT
+    // Prevent other threads from hijacking the Exception filter, and allocate 4MB up-front that may be useful in
+    // low-memory scenarios.
+    DWORD bugSplatFlags = MDSF_PREVENTHIJACKING | MDSF_USEGUARDMEMORY;
+    MiniDmpSender mpSender { "interface_alpha", "Interface", BuildInfo::VERSION.toLatin1().constData(), nullptr, bugSplatFlags };
```
Comment (birarda) [inline interface/src/main.cpp]: constantize these options

**Comment by hifi-gustavo** [comment]
Build console output for commit: d2264163067117d51727b2275128d93204ec1ae2 available at: https://gist.github.com/a8e1ed8dcc8614275f3d<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7052.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7052.dmg

**Pull request merged**